### PR TITLE
Fix #947 and some other NSProgress incongruities.

### LINKED ISSUE #947: [NSProgress] fractionCompleted does not fire KVO when it changes
It is missing a dependency on completedUnitCount.

### PR DESCRIPTION
**Pull request by DHowett-MSFT**
NSProgress should fire KVO in a lot more cases than it did. Now it does!

---

This change is [Reviewable](https://reviewable.io/reviews/microsoft/winobjc/1054)

### CODE REVIEW

**Comment by rajsesh-msft** [comment]
Is there test collateral associated with this change?

**Comment by memontic-ms** [comment]
:shipit:

**Comment by ms-jihua** [comment]
:shipit:

**Comment by DHowett-MSFT** [comment]
This is currently tested by one of our middleware components, but I _can_ add some tests here.

My concern is that we are growing a significant number of ad-hoc KVO tests, each with a custom implementation, and they need to be unified via a common listener to avoid duplicating the same code over and over in different test facades.

---
In reply to: [250597295](https://github.com/Microsoft/WinObjC/pull/1054#issuecomment-250597295) [](ancestors = 250597295)

**Comment by rajsesh-msft** [comment]
This should be part of NSProgress tests.  Part of our long term plan to improve test code coverage is to not check-in code without tests.  I am not worried about duplicating test code.

---
In reply to: [250880931](https://github.com/Microsoft/WinObjC/pull/1054#issuecomment-250880931) [](ancestors = 250880931,250597295)

**Comment by bbowman** [comment]
+100 here.

---
In reply to: [250881581](https://github.com/Microsoft/WinObjC/pull/1054#issuecomment-250881581) [](ancestors = 250881581,250880931,250597295)

**Comment by bbowman** [comment]
:clock1:

**Comment by DHowett-MSFT** [comment]
@rajsesh-msft  @bbowman test collateral added! finally ;P

**Comment by jaredhms** [comment]
>             child->_parentPendingUnitCount = inUnitCount;

does child not need to fire KVO for these?

---
Refers to: Frameworks/Foundation/NSProgress.mm:194 in b2ed69d. [](commit_id = b2ed69da0fd228666504865e533cc40588e0eddd, deletion_comment = False)

File: tests/unittests/Foundation/NSProgressTests.mm [hunk 1]
```diff
@@ -123,8 +125,15 @@ - (void)addParentTo:(NSProgress*)child {
     ASSERT_EQ(0, [userInfo totalUnitCount]);
     ASSERT_EQ(0, [userInfo completedUnitCount]);
 
-    [userInfo setTotalUnitCount:2];
-    [userInfo setCompletedUnitCount:1];
+    _NSFoundationTestKVOFacade* kvoListener = [[_NSFoundationTestKVOFacade newWithObservee:userInfo] autorelease];
+    [kvoListener observeKeyPath:@"fractionCompleted"
```
Comment (jaredhms) [inline tests/unittests/Foundation/NSProgressTests.mm]: >bserveKeyPath:@"fractionCompleted" [](start = 18, length = 34)

worth adding an 'indeterminate' observer too?

**Comment by rajsesh-msft** [comment]
:shipit:

**Comment by jaredhms** [comment]
:shipit:

**Pull request merged**